src/test/name_applications_tests.cpp

```diff
@@ -119,4 +119,12 @@ BOOST_AUTO_TEST_CASE( name_description )
     BOOST_CHECK(DescFromName(identityValid, NameNamespace::Identity) == "'id/wikileaks'");
 }
 
+BOOST_AUTO_TEST_CASE( valid_json )
```

Comment:
Now that the empty string is special-cased in the implementation, can we have a test case here for that?

Reply:
Oh wait, I just realized that special case didn't get moved like I suggested. Can that be moved? I think it would make the code cleaner.